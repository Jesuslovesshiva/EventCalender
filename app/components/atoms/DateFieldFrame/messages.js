import {defineMessages} from 'react-intl'

const scope = 'atoms.date-field-frame'

export default defineMessages({
  toggle: {
    id: `${scope}.toggle`,
    defaultMessage: 'Kalendar ein-/ausschalten',
    description: 'aria label toggle calendar button'
  }
})
