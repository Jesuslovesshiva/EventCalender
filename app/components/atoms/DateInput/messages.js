import {defineMessages} from 'react-intl'

const scope = 'atoms.date-picker'

export default defineMessages({
  ddmmyyyy: {
    id: `${scope}.date`,
    defaultMessage: 'TT.MM.JJJJ',
    description: 'placeholder for format dd.MM.yyyy'
  }
})
