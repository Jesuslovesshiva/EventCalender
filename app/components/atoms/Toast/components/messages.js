import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
