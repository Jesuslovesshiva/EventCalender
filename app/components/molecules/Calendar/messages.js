import {defineMessages} from 'react-intl'

const scope = 'molecules.calendar'

export default defineMessages({
  selectMonth: {
    id: `${scope}.selectMonth`,
    defaultMessage: 'Monat auswählen'
  },
  selectYear: {
    id: `${scope}.selectYear`,
    defaultMessage: 'Jahr auswählen'
  },
  nextMonth: {
    id: `${scope}.nextMonth`,
    defaultMessage: 'nächstes Monat',
    description: 'aria label nextMonth button'
  },
  prevMonth: {
    id: `${scope}.prevMonth`,
    defaultMessage: 'vorheriges Monat',
    description: 'aria label prevMonth button'
  }
})
