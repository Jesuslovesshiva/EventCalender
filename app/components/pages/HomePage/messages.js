import {defineMessages} from 'react-intl'

const scope = 'app'

export default defineMessages({
  someMessage: {
    id: `${scope}.someMessage`,
    defaultMessage: 'This is a message to show localization usage.',
  }
})
