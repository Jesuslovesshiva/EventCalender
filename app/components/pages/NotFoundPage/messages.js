import {defineMessages} from 'react-intl'

const scope = 'app'

export default defineMessages({
  notFound: {
    id: `${scope}.notFound`,
    defaultMessage: '404 - not found',
    description: 'not found message'
  }
})
