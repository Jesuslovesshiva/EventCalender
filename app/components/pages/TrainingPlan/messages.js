import {defineMessages} from 'react-intl'

const scope = 'pages.trainings-plan'

export default defineMessages({
  pageTitle: {
    id: `${scope}.pageTitle`,
    defaultMessage: 'Trainingsplan',
    description: 'title of the page'
  },
  training: {
    id: `${scope}.training`,
    defaultMessage: 'Training'
  },
  homeGame: {
    id: `${scope}.homeGame`,
    defaultMessage: 'Heimspiel'
  },
  roadGame: {
    id: `${scope}.roadGame`,
    defaultMessage: 'Auswärtsspiel'
  },
  other: {
    id: `${scope}.other`,
    defaultMessage: 'Sonstiges'
  },
  tournament: {
    id: `${scope}.tournament`,
    defaultMessage: 'Turnier'
  },
  exercises: {
    id: `${scope}.exercises`,
    defaultMessage: 'Übungen'
  },
  loadingError: {
    id: `${scope}.loadingError`,
    defaultMessage: 'Was not able to load events. Try again later.'
  }
})
