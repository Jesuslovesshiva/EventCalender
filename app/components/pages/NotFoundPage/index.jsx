import {useRouter} from '@tanstack/react-location'
import {useMemo} from 'react'
import {FormattedMessage} from 'react-intl'

import dummyMenu from '@/components/molecules/Navigation/dummy-menu'
import PageHeader from '@/components/molecules/PageHeader'

import messages from './messages'

const traverseMenu = (pathParts, menu = dummyMenu) => {
  const menuItem = menu.find(item => item.path === pathParts[0])
  const pathPartsRest = pathParts.slice(1)

  if (menuItem) {
    if (pathPartsRest.length) {
      return traverseMenu(pathPartsRest, menuItem.children ?? [])
    }

    return menuItem
  }

  return null
}

const NotFoundPage = () => {
  const router = useRouter()
  const pathParts = useMemo(
    () => router.state.location.pathname.split('/').filter(Boolean),
    [router.state.location]
  )
  const activeMenuItem = traverseMenu(pathParts)

  if (activeMenuItem?.label) {
    return <PageHeader title={activeMenuItem.label} />
  }

  return (
    <div className="flex justify-center items-center h-full font-page-heading text-6xl text-primary">
      <FormattedMessage {...messages.notFound} />
    </div>
  )
}

export default NotFoundPage
