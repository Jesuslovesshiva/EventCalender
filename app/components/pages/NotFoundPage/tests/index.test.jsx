import {act} from '@testing-library/react'

import {history} from '@/configure-store'
import render from '@/utils/test-utils/custom-render'

import NotFoundPage from '..'
import messages from '../messages'

const renderComponent = () =>
  render(<NotFoundPage />, {wrapperProps: {hasRouter: true}})

describe('NotFoundPage', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should render 404 message on unknown page', () => {
    const {getByText} = renderComponent()
    act(() => {
      history.push('/training-plan/periodization/foo')
    })
    expect(getByText(messages.notFound.defaultMessage)).toBeInTheDocument()
  })

  it('should render page title if found in dummy menu', () => {
    const {getByText} = renderComponent()
    act(() => {
      history.push('/training-plan/periodization')
    })
    expect(getByText('Periodization')).toBeInTheDocument()
  })
})
