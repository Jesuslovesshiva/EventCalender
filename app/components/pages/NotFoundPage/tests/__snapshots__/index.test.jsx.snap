// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should match the snapshot 1`] = `
<div>
  <div
    class="flex justify-center items-center h-full font-page-heading text-6xl text-primary"
  >
    404 - not found
  </div>
</div>
`;
